src/gui/editors/SongEditor.cpp

```diff
@@ -839,15 +896,45 @@ void SongEditor::updatePositionLine()
 
 void SongEditor::zoomingChanged()
 {
-	setPixelsPerBar( m_zoomLevels[m_zoomingModel->value()] * DEFAULT_PIXELS_PER_BAR );
+	setPixelsPerBar(static_cast<int>(m_zoomingLogModel->value() / 100.0f * DEFAULT_PIXELS_PER_BAR));
 
 	m_song->m_playPos[Song::Mode_PlaySong].m_timeLine->
 					setPixelsPerBar( pixelsPerBar() );
 	realignTracks();
 	updateRubberband();
 	m_timeLine->setSnapSize(getSnapSize());
 	
-	emit zoomingValueChanged( m_zoomLevels[m_zoomingModel->value()] );
+	emit zoomingValueChanged(m_zoomingLogModel->value() / 100.0f);
+}
+
+
+int SongEditor::scaleFunction(int val, bool exp)
```

Comment:
This should be split in two separate functions. They don't have to be part of the `SongEditor` class. Since they rely on the `ZOOM_` constants the naming should give us a hint about that, like `linearZoom()` and `expZoom()`

Reply:
I agree, two functions are better. I'll do that. 
But what do you mean about "They don't have to be part to the SongEditor class"?

Reply:
> But what do you mean about "They don't have to be part to the SongEditor class"?

The two functions don't need to be in the `SongEditor` class. They are independent from the class itself and can be reused elsewhere.

Reply:
> They are independent from the class itself

This means you can create a regular old-school function and put it at the bottom of `SongEditor.cpp` like so:

```cpp
int scaleFunction(int val, bool exp)
{
    ...
}
```

> can be reused elsewhere

Well, since they are kind of adapted to Song Editor I would just let them be local (unless you find somewhere to reuse them). This means you **don't** declare the functions in `SongEditor.h`.

Reply:
Not sure about how to do that without declaring the function in SongEditor.h  Can you point me to another class that has something similar?

Reply:
Put it at the top of the file and drop the `SongEditor::` part. :)

Reply:
I was following "..and put it at the bottom of ..." Now it works! Thanks!

Reply:
Ah, the order matters, sorry :eyes:

Reply:
Done.